### PR TITLE
105465 school finances loans

### PR DESCRIPTION
**Pull request by MarkRobinsonDfe**
just a very beta draft

### CODE REVIEW

**Review by wystanpalm-madetech — COMMENTED** [review]
A few comments here, some minor but I think there's a significant change to make moving the loan reference to the ISchool.

File: Dfe.Academies.Academisation.Domain.Core/ApplicationAggregate/SchoolDetails.cs [hunk 1]
```diff
@@ -12,6 +12,8 @@ public record SchoolDetails(
 	// finances
 	FinancialYear CurrentFinancialYear,
 	FinancialYear NextFinancialYear,
+	// leases & loans
+	Dictionary<int, LoanDetails> Loans,
```
[the remark this answers is not in the document]
Comment (MarkRobinsonDfe) [reply Dfe.Academies.Academisation.Domain.Core/ApplicationAggregate/SchoolDetails.cs:16]: Yes, this was one of the Q's I wasn't sure about i.e. why we were using Dictionary's so wanted to chat through the reasons why @wystanpalm-madetech

File: Dfe.Academies.Academisation.Data/AcademisationContext.cs [hunk 2]
```diff
@@ -42,9 +43,21 @@ public override int SaveChanges()
 			{
 				continue;
 			}
-
+			
 			foreach (object? child in children)
 			{
+				Type childType = child.GetType();
```
Comment (wystanpalm-madetech) [inline Dfe.Academies.Academisation.Data/AcademisationContext.cs:49]: Must make this more generic -- but not in this PR, it's big enough already.

**Review by wystanpalm-madetech — APPROVED** [review]
Good effort Mark, this turned out to me much more tricky than I was expecting. 

Please do rework the AcademisationContext.SaveChanges to handle multiple levels of subobject in a generic way, just not pushing back on this PR for it because I think it's better to get it merged.

**Review by wystanpalm-madetech — APPROVED** [review]

File: Dfe.Academies.Academisation.Data/AcademisationContext.cs (hunk 2 above)
Comment (MarkRobinsonDfe) [reply Dfe.Academies.Academisation.Data/AcademisationContext.cs:49]: yeah, was just trying to get this working @wystanpalm-madetech , definately a hacky solution. Will probably get @paullocknimble to look into this and refactor

**Pull request merged**